File: nodejs/api_artifacts.js
/**
 * This script contains an example of checking channels for artifacts in a
 * Petal Metrics API call. You will need a valid developer API key to access.
 *
 * Usage: node api_artifacts.js -k $API_KEY
 */
const { Command } = require('commander');
const { requestMetrics } = require('./api');

// parse API key from command line args
const program = new Command();
program
  .requiredOption('-k, --api-key <key>', 'Petal Metrics API key');
program.parse(process.argv);
const { apiKey } = program;

// construct a random EEG data array
let randomEEGData = Array(4);
for (let i = 0; i < randomEEGData.length; i++) {
  randomEEGData[i] = Array.from({ length: 256 }, () =>
    Math.floor(Math.random() * 10)
  );
}

requestMetrics(
  apiKey,
  randomEEGData,
  ["artifact_count"]
)
  .then(calculations => {
    console.log(calculations.artifact_count);
  });

File: python/api_artifacts.py
'''
This script contains an example of checking channels for artifacts in a Petal
Metrics API call. You will need a valid developer API key to access.

Usage: python api_artifacts.py -k $API_KEY
'''
import argparse
import pprint
import random

import api


parser = argparse.ArgumentParser()
parser.add_argument('-k', '--api_key', type=str, required=True,
                    help='API key for the Petal Metrics API')
args = parser.parse_args()

random_eeg_data = [
    [random.randint(0,100) / 100 for i in range(256)]
    for num in range(4)
]
calculations = api.request_metrics(
    api_key=args.api_key,
    eeg_data=random_eeg_data,
    metrics=['artifact_count'],
)
pprint.pprint(calculations['artifact_count'])
